docs/DevEnvironment.md

```diff
@@ -0,0 +1,158 @@
+<!--
+# @title Development Environment
+-->
+# Development Environment
+
+## Introduction
+
+Rouge is written in Ruby and has a number of development dependencies. To
+develop new features for Rouge (like a lexer for a new syntax) or to fix bugs,
+you need to set up a development environment.
+
+### Ruby and Git
+
+First thing's first. This guide is _not_ a guide to installing Ruby and Git.
+There are a number of excellent resources out there explaining how to do these
+things. For Ruby, we recommend [the official documentation][rb-inst-docs] and
+for Git, [GitHub's documentation][gh-inst-docs].
+
+[rb-inst-docs]: https://www.ruby-lang.org/en/documentation/installation/
```

Comment:
It looks good, do you think it's worth adding something about virtual environments or is that too complex for the target audience?

Reply:
You mean like environments in a virtual machine? Or containers like Docker? Or something else?

Reply:
Something like https://github.com/rbenv/rbenv

Reply:
Ah, yeah, I don't think it's worth trying to include that. If you're using Bundler to install the gems to `vendor/` that should really solve most problems.

There's also just so many choices: `rbenv`, `rvm`, `chruby`, `asdf`, `ry`, etc. I'd rather leave installation as something for the user to decide on their own.